bam/runtime/src/main/java/io/quarkiverse/langchain4j/bam/runtime/config/Langchain4jBamConfig.java

```diff
@@ -75,12 +75,28 @@ interface BamConfig {
         Boolean logResponses();
 
         /**
-         * Whether or not to enable the integration. Defaults to {@code true}, which means requests are made to the OpenAI
-         * provider.
+         * Whether or not to enable the chat model integration. Defaults to {@code true}, which means requests are made to the
+         * BAM provider.
          * Set to {@code false} to disable all requests.
          */
         @WithDefault("true")
-        Boolean enableIntegration();
+        Boolean enableChatModel();
+
+        /**
+         * Whether or not to enable the streaming chat model integration. Defaults to {@code true}, which means requests are
+         * made to the BAM provider.
+         * Set to {@code false} to disable all requests.
+         */
+        @WithDefault("true")
+        Boolean enableStreamingChatModel();
+
+        /**
+         * Whether or not to enable the embedding model integration. Defaults to {@code true}, which means requests are made to
+         * the BAM provider.
+         * Set to {@code false} to disable all requests.
+         */
+        @WithDefault("true")
+        Boolean enableEmbeddingModel();
```

Comment:
Yes, because I need to have a "fake" implementation of the StreamingChatModel for the module that doesn't implement it, otherwise I will get an exception during the build phase from the [AiServicesProcessor](https://github.com/quarkiverse/quarkus-langchain4j/commit/bf2054ef7bd33e3ab3fc30e46b400978d12c3afd#diff-f9271ecf02b95bde657540107625a1f1c101a4c45c4fe5987ac1a503cad85d1f) class

Reply:
Maybe there is a faster way to have the same behavior 😆

Reply:
> Yes, because I need to have a "fake" implementation of the StreamingChatModel for the module that doesn't implement it, otherwise I will get an exception during the build phase from the [AiServicesProcessor](https://github.com/quarkiverse/quarkus-langchain4j/commit/bf2054ef7bd33e3ab3fc30e46b400978d12c3afd#diff-f9271ecf02b95bde657540107625a1f1c101a4c45c4fe5987ac1a503cad85d1f) class

Hm... The way I see if, `Multi<String>` should not be an allowed return option unless a `StreamingChatModel` exists.
I don't see how this property makes a difference, but I might be missing something.

Reply:
I think I understand what you are saying, if a method of the interface annotated with `@RegisterAiService` returns a `Multi<String>` and there is no implementation it is possible to generate an error. Maybe this is a "smarter" way to go 😆

Reply:
Do you have the stacktrace?

I am still not convinced this change is really necessary for the functionality this great PR brings to the table.

Reply:
> Is there a good reason why we now have a specific property for each one of the models?

The ability to disable a particular model was added to upstream Langchain4j, so here in the config, each model has a runtime property which can disable it if needed/wanted. It enables/disables the integration as a whole.

Its really intended to be able to disable everything and not make an actual call if the application deems not to (like perhaps its deployed in an environment where they don't have access to a key and don't want to make the call across the wire).

Reply:
> Is there any chance you can update the PR to have a commit that reverts the change to the `enable-integrarion` config properties? I would like to try it out myself

Now there are only the changes done to the BAM and CORE modules

Reply:
@geoand let me know if you solve the problem in another way or if I have to recommit the code related to the `enableStreamingChatModel()`

Reply:
Thanks, I'll have a look